### PR TITLE
Remove six dependency

### LINKED ISSUE #1926: Remove 'six' bundled dependency
The `six` package is bundled in vispy here: vispy/ext/_bundled/six.py
And imported here: vispy/ext/six.py

This should be removed and *all* usage of the `six` library removed/replaced with python 3 equivalent checks throughout vispy.

### PR DESCRIPTION
**Pull request by sgaist**
This merge request removes the bundled six library and updates the relevant parts to Python 3

Fixes #1926

### CODE REVIEW

File: vispy/app/application.py [hunk 1]
```diff
@@ -106,7 +105,9 @@ def is_interactive(self):
 
         # IPython does not set sys.flags when -i is specified, so first
         # check it if it is already imported.
-        if '__IPYTHON__' not in dir(six.moves.builtins):
+        try:
+            get_ipython()
+        except NameError:
```
Comment (djhoese) [inline vispy/app/application.py:110]: Are you sure this behaves the same? Wouldn't this work:

```
import builtins
if hasattr(builtins, '__IPYTHON__'):
```

Maybe these are essentially the same?

File: vispy/testing/image_tester.py [hunk 2]
```diff
@@ -49,9 +49,9 @@
 from subprocess import check_call, CalledProcessError
 import numpy as np
 
-from ..ext.six import string_types
-from ..ext.six.moves import http_client as httplib
-from ..ext.six.moves import urllib_parse as urllib
+import http.client as httplib
```
Comment (djhoese) [inline vispy/testing/image_tester.py:52]: From what I can tell, this module only uses HTTPConnection from http.client and urlencode from urllib.parse, mind just changing these to import those directly?

**Review by djhoese — CHANGES_REQUESTED** [review]
Nice job! What a huge PR. Thanks for doing all this work. I had two small comments (one question, one change request). Otherwise this looks good.

File: vispy/app/application.py (hunk 1 above)
Comment (sgaist) [reply vispy/app/application.py:110]: AFAIK, it should however you are correct, we can keep the original logic. I somehow messed up something in my installation when doing the checks to update the code.

File: vispy/testing/image_tester.py (hunk 2 above)
Comment (sgaist) [reply vispy/testing/image_tester.py:52]: No problem with that

**Review by djhoese — APPROVED** [review]
Thanks!

File: vispy/util/fetching.py [hunk 3]
```diff
@@ -12,8 +12,8 @@
 import shutil
 import time
 
-from ..ext.six.moves import urllib
-from ..ext.six import string_types
+import urllib
```
[the remark this answers is not in the document]
Comment (djhoese) [reply vispy/util/fetching.py:15]: I only see a `urllib.request` usage so doing `import urllib.request` should work.

[the remark this answers is not in the document]
Comment (sgaist) [reply vispy/util/fetching.py:15]: Indeed, that's the correct fix.

**Comment by djhoese** [comment]
Looks like there are a couple examples that are using six too:

```
Example /basics/scene/flipped_axis.py failed (1):

----------------------------------------------------------------------

Traceback (most recent call last):

  File "<string>", line 12, in <module>

  File "/home/travis/build/vispy/vispy/examples/basics/scene/flipped_axis.py", line 25, in <module>

    from vispy.ext.six import next

ModuleNotFoundError: No module named 'vispy.ext.six'
```

**Comment by sgaist** [comment]
My search-fu was bad on these two... Fixed

**Comment by djhoese** [comment]
I think you forgot to push those fixes :wink:

**Comment by sgaist** [comment]
Almost, unlocking the key failed and I missed that 😅

**Comment by djhoese** [comment]
Travis notification is stuck, but is passing. I think this all looks good. Thanks!

**Pull request merged**